test/classes/initializers/assign-param-segfault.chpl

```diff
@@ -1,19 +0,0 @@
-enum Foo { Foo0, Foo1, Foo2, Foo3 };
-enum Baz { Baz0, Baz1, Baz2, Baz3 };
-class Bar {
-  param foo: Foo;
-  var baz: Baz;
-  proc init (param foo: Foo = Foo.Foo0) {
-    this.foo = foo;
-    select foo {
-      when Foo.Foo0 do this.baz = Baz.Baz0;
-      when Foo.Foo1 do this.baz = Baz.Baz1;
-      when Foo.Foo2 do this.baz = Baz.Baz2;
-      when Foo.Foo3 do this.baz = Baz.Baz3;
-      otherwise this.baz = Baz.Baz0;
-      }
-  }
-}
-var bar = new Bar(Foo.Foo1);
-writeln(bar);
-delete bar;
```

Comment:
Please explain in the PR message why this test is going away.

It does not seem to fall in the category "tested a bug in constructors" (even though it may do).

Ditto `initializeMethod.chpl` below.

Reply:
Oops, I think I meant to remove the variant of this test under constructors/ , I'll add it back in.